godot-macros/src/derive_godot_class.rs

```diff
@@ -265,51 +304,109 @@ fn make_deref_impl(class_name: &Ident, fields: &Fields) -> TokenStream {
+
+        let ExportHint {
+            hint_type,
+            description,
+        } = exported_field.hint.clone().unwrap_or_else(ExportHint::none);
+
+        // trims '"' and '\' from both ends of the hint description.
+        let description = description.trim_matches(|c| c == '\\' || c == '"');
+
+        let getter_name;
+        match &exported_field.getter {
+            GetterSetter::Omitted => {
+                getter_name = "".to_owned();
+            }
+            GetterSetter::Generated => {
+                getter_name = format!("get_{field_name}");
+                let getter_ident = ident(&getter_name);
+                let signature = quote! {
+                    fn #getter_ident(&self) -> #field_type
+                };
+                getter_setter_impls.push(quote! {
+                    pub #signature {
+                        self.#field_ident
+                    }
+                });
+                export_tokens.push(quote! {
+                    ::godot::private::gdext_register_method!(#class_name, #signature);
+                });
+            }
+            GetterSetter::Custom(name) => {
+                getter_name = name.clone();
+                let getter_ident = ident(&getter_name);
+                export_tokens.push(existence_check(&getter_ident));
+            }
+        }
+
+        let setter_name;
+        match &exported_field.setter {
+            GetterSetter::Omitted => {
+                setter_name = "".to_owned();
+            }
+            GetterSetter::Generated => {
+                setter_name = format!("set_{field_name}");
+                let setter_ident = ident(&setter_name);
+                let signature = quote! {
+                    fn #setter_ident(&mut self, #field_ident: #field_type)
+                };
+                getter_setter_impls.push(quote! {
+                    pub #signature {
+                        self.#field_ident = #field_ident;
+                    }
+                });
+                export_tokens.push(quote! {
+                    ::godot::private::gdext_register_method!(#class_name, #signature);
+                });
+            }
+            GetterSetter::Custom(name) => {
+                setter_name = name.clone();
+                let setter_ident = ident(&setter_name);
+                export_tokens.push(existence_check(&setter_ident));
+            }
+        };
+
+        export_tokens.push(quote! {
+            use ::godot::builtin::meta::VariantMetadata;
+
+            let class_name = ::godot::builtin::StringName::from(#class_name::CLASS_NAME);
+
+            let property_info = ::godot::builtin::meta::PropertyInfo::new(
+                <#field_type>::variant_type(),
+                ::godot::builtin::meta::ClassName::of::<#class_name>(),
+                ::godot::builtin::StringName::from(#field_name),
+                ::godot::engine::global::PropertyHint::#hint_type,
+                GodotString::from(#description),
```

Comment:
`GodotString` should be qualified too.

In general, if the resulting code is locally scoped, it's also fine to `use` these identifiers if it makes resulting code more readable. I'll leave that up to you 🙂

Reply:
Done.